File: Ejercicios_Varios/Java/Proyecto_Java/src/main/java/Ejecicios/Ejercicio_2.java
package Ejecicios;

import java.util.Scanner;

public class Ejercicio_2 {
    /*Escribir un programa  que al escribir como dato un numero
    entero positivo N, Calcule el resultado de las siguientes series:
    1 + (1/2) + (1/3) +(1/4) ... (1/N)
    Si el usuario escribe un numero incorrecto, el programa no se ejecuta
    En cambio, pregunta de nuevo spor la informacion hasta que el dato sea correcto

     */
    public static void main(String[] args) {

        while (true) {
            Scanner input = new Scanner(System.in);
            System.out.println("Ingrese un numero entero positivo: ");
            int numero = input.nextInt();
            float resultado= 0;

            if (numero > 0) {
                System.out.println("el valor que se ingreso es correcto");
                for (int i=1;i<=numero; i++ ){

                    resultado = resultado +(1/(float)i);
                    System.out.println(" el resultado de es: "+ resultado);
                    {break;} // cierra el buvle infinito del while

                }


            } else {
                System.out.println("Error---> el numero (" + numero + ") no es entero");
            }
        }
    }

}

File: Ejercicios_Varios_2/Python/CursoPython/Ejercicios/Ejercicio2.py
'''

    Escribe un programa que permita generar la tabla de un numero entero
    positivo N, comenzando desde 1.

    Si el usuario escribe un numero incorrecto el programa no se ejecuta , en cambio preguntara de nuevo por la informacion
    hasta que el dato ingresado sea correcto

'''

comparar = True

while  comparar==True:

    numero1 = int(input("Ingrese numero entero positivo: \n"))

    if numero1 > 0:
        print("el numero ingresado es correcto!! \n")

        for i in range(1, 11):
            print(f'{numero1} X {i} = {numero1 * i}')

            comparar = False
    else:
        print(f'el numero ingresado no es correcto, intente de nuevo')